napari/utils/misc.py

```diff
@@ -463,23 +465,44 @@ def dir_hash(path: Union[str, Path], include_paths=True, ignore_hidden=True):
             )
         )
 
-    _hash = hashfunc()
+    hash_func = hashlib.md5
+    _hash = hash_func()
     for root, _, files in os.walk(path):
         for fname in sorted(files):
             if fname.startswith(".") and ignore_hidden:
                 continue
-            # update the hash with the file contents
-            file = Path(root) / fname
-            _hash.update(file.read_bytes())
+            _file_hash(_hash, Path(root) / fname, path, include_paths)
+    return _hash.hexdigest()
 
-            if include_paths:
-                # update the hash with the filename
-                fparts = file.relative_to(path).parts
-                _hash.update(''.join(fparts).encode())
 
+def paths_hash(
+    paths: Iterable[Union[str, Path]],
+    include_paths: bool = True,
+    ignore_hidden: bool = True,
+) -> str:
+    """Compute the hash of list of paths."""
+    import hashlib
+
+    hash_func = hashlib.md5
+    _hash = hash_func()
+    for file_path in sorted(paths):
+        file_path = Path(file_path)
+        if ignore_hidden and str(file_path.stem).startswith("."):
+            continue
+        _file_hash(_hash, file_path, file_path.parent, include_paths)
     return _hash.hexdigest()
 
 
+def _file_hash(_hash, file: Path, path: Path, include_paths: bool = True):
```

Comment:
Can you add a "parameters" section to the doc string here

Reply:
Done